content/blog/2021-11-17-exp-versioning.md

```diff
@@ -0,0 +1,243 @@
+(forgetting to log a particular value or a whole experiment, or writing down the
+wrong value). Organizing and comparing experiments was limited to sorting
+spreadsheets. Reproducing experiments was still left to traditional version
+control, if that was even used.
+
+Experiment tracking tools were developed to better address experiment tracking
+needs. They provide you with an API to log experiment information, a database to
+store it, and a dashboard to compare and visualize. You can log all types of
+artifacts, label and compare experiments in a dashboard, and collect logged
+artifacts to reproduce the experiment.
+
+However, as projects grow, these experiment tracking tools leave a lot up to you
+and start to feel cumbersome:
+
+- You need to decide what is important to log. Much of the information is the
+  same between experiments, and it's not clear what changed in each.
+- You need to keep experiments organized, you need to manage who has access to
+  them, and you may need to drop experiments that are outdated or otherwise
+  irrelevant.
+- You need to set up a process to reproduce an experiment from the logged
+  artifacts, running it again from scratch. If any input is not logged,
+  reproducibility is lost.
+
+# Experiment Versioning in DVC
+
+How does DVC help?
+
+- Log experiments end to end (commands, inputs, and outputs) and track changes
+  introduced by each experiment.
+- Associate experiments with Git commits, branches, and tags from which they
+  derive. Organize locally and choose what to share.
+- Reproduce any previous experiment with a single command. Cache previous runs
+  and results to reduce duplicate compute time and storage space.
+
+We believe this is the next generation of tracking experiments. Let's look at a
+history of version control (from
+https://ericsink.com/vcbe/html/history_of_version_control.html):
+
+| Generation | Networking  | Operations         | Concurrency         |
+| ---------- | ----------- | ------------------ | ------------------- |
+| First      | None        | One file at a time | Locks               |
+| Second     | Centralized | Multi-file         | Merge before commit |
+| Third      | Distributed | Changesets         | Commit before merge |
+
+See also https://git-scm.com/book/en/v2/Getting-Started-About-Version-Control.
+
+Now let's compare to experiment tracking:
+
+| Generation | Platform                   | Logging        | Organization | Reproducibility | Collaboration |
+| ---------- | -------------------------- | -------------- | ------------ | --------------- | ------------- |
+| First      | Spreadsheet, notebook      | Scalar values  | None         | None            | None          |
+| Second     | API + database + dashboard | Multi-artifact | Labels       | Manual          | Centralized   |
+| Third      | DVC + Git                  | Changesets     | Git-based    | Automatic       | Distributed   |
```

Comment:
`DVC + Git` - there are no other names but DVC. It might be better to include other names or not to use DVC.
Git is fine I think.